### PR TITLE
fix(ignore): Fix 2 to 3 migration enabling frontend/availabilty/homeassistant when not set

### PR DESCRIPTION
**Pull request by Koenkk**
Fix issue with the 2 to 3 migration enabling features when not set, e.g. before this PR:

```yaml
homeassistant: false
```

would result in

```yaml
homeassistant:
  enabled: false
frontend:
  enabled: true
availability:
  enabled: true
```

This PR fixes that

### CODE REVIEW

**Comment by Nerivec** [comment]
The problem appears to be in `getValue` rather (`validPath` should be false):

https://github.com/Koenkk/zigbee2mqtt/blob/329b8c9c27c2ba2537b59ef16241a6e84167fa7b/lib/util/settingsMigration.ts#L90-L92

```ts
        if (i === path.length - 1) {
            const setting = currentSettings[key];

            return [setting != undefined, setting];
        } else {
```

That should also fix the improper log in your test (should only log `homeassistant`).
```diff
            expect(migrationNotesContent).toContain(`[SPECIAL] Property 'homeassistant' is now always an object.`);
-            expect(migrationNotesContent).toContain(`[SPECIAL] Property 'frontend' is now always an object.`);
-            expect(migrationNotesContent).toContain(`[SPECIAL] Property 'availability' is now always an object.`);
```

**Comment by Koenkk** [comment]
@Nerivec I think I fixed it correctly now.

File: lib/util/settingsMigration.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ function getValue(currentSettings: any, path: string[]): [validPath: boolean, va
     for (let i = 0; i < path.length; i++) {
         const key = path[i];
 
-        if (i === path.length - 1) {
+        if (i === path.length - 1 && currentSettings[key] !== undefined) {
```
Comment (Nerivec) [inline lib/util/settingsMigration.ts:89]: That'll force it to go into the `else` though, and re-lookup/re-test again.

Comment (Koenkk) [reply lib/util/settingsMigration.ts]: True, improved it.

File: lib/util/settingsMigration.ts [hunk 2]
```diff
@@ -85,16 +85,21 @@ function setValue(currentSettings: any, path: string[], value: unknown, createPa
...
         const key = path[i];
+        const value = currentSettings[key];
 
         if (i === path.length - 1) {
-            return [true, currentSettings[key]];
+            if (value == undefined) {
+                return [false, undefined];
+            } else {
+                return [true, value];
+            }
```
Comment (Nerivec) [inline lib/util/settingsMigration.ts:95]:
```ts
return [value != undefined, value];
```

`value` is expected not be be used if `validPath=false`, so, shouldn't be a problem if it's null, the double-eq will set `validPath` to false.

Comment (Koenkk) [reply lib/util/settingsMigration.ts]: done!

**Pull request merged**